### PR TITLE
wg-k8s-infra: canary prowjobs for sig-scalability

### PR DESCRIPTION
**Pull request by ameukam**
Add canary jobs running on `k8s-infra-prow-build` for some sig-scalability periodics prowjobs.
Switch from `--gcp-project=foo` to `--gcp-project=k8s-infra-e2e-scale-5k-project`

### CODE REVIEW

**Comment by ameukam** [comment]
xref: https://github.com/kubernetes/k8s.io/issues/1469. 
Failures are expected so there is no need of a active babysitting.

File: config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+      - --env=CONCURRENT_SERVICE_SYNCS=5
+      - --env=HEAPSTER_MACHINE_TYPE=e2-standard-32
+      - --extract=ci/latest-fast
+      - --extract-ci-bucket=k8s-release-dev
+      # Overrides CONTROLLER_MANAGER_TEST_ARGS from preset-e2e-scalability-periodics.
+      - --env=CONTROLLER_MANAGER_TEST_ARGS=--profiling --kube-api-qps=100 --kube-api-burst=100 --endpointslice-updates-batch-period=500ms
+      - --gcp-master-image=gci
+      - --gcp-node-image=gci
+      - --gcp-node-size=e2-small
+      - --gcp-nodes=5000
```
Comment (spiffxp) [inline config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml:35]: We'll need a quota request for this.  I'd rather avoid giving all scalability projects this kind of quota, we should pin to a specific project for this.

File: config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+      - /workspace/scenarios/kubernetes_e2e.py
+      args:
+      - --cluster=gce-scale-cluster
+      - --env=HEAPSTER_MACHINE_TYPE=e2-standard-32
+      # TODO(mborsz): Adjust or remove this change once we understand coredns
+      # memory usage regression.
+      - --env=KUBE_DNS_MEMORY_LIMIT=300Mi
+      - --extract=ci/latest-fast
+      - --extract-ci-bucket=k8s-release-dev
+      - --gcp-nodes=5000
```
Comment (spiffxp) [inline config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml:95]: Same. This should use the same project as correctness.  What's the node type used here?

File: config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+    - image: gcr.io/k8s-testimages/kubekins-e2e:v20210601-ea6aa4e-master
+      command:
+      - runner.sh
+      - /workspace/scenarios/kubernetes_e2e.py
+      args:
+      - --cluster=kubemark-5000
+      - --extract=ci/latest
+      - --gcp-node-image=gci
+      - --gcp-node-size=e2-standard-8
+      - --gcp-nodes=84
```
Comment (spiffxp) [inline config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml:175]: Honestly not sure if this will need quota increase or not

**Review by spiffxp — COMMENTED** [review]
/lgtm
/hold
If you expect these to fail and just want to see how, then remove the hold.  But I'm not ok with the 5k node jobs as-is.

I would much rather see provisioning of a special 5k node project for the 5k node jobs.  Like modify `infra/gcp/prow/ensure-e2e-projects.sh` to add a `k8s-infra-e2e-scale-5k-project` to `E2E_MANUAL_PROJECTS` and then pin the jobs to that.

File: config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml (hunk 2 above)
Comment (ameukam) [reply config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml:95]: Took a lot at a successful job of  `ci-kubernetes-e2e-gce-scale-performance`: https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-scale-performance/1401585162642264064/build-log.txt. 
The node type appears to be e2-standard-32.

File: config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml (hunk 3 above)
Comment (ameukam) [reply config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml:175]: No need to increase the quota for this job. the smallest quota in us-east1 is 1250 (for all the projects with type `scalability-project`)

File: config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+      - runner.sh
+      - /workspace/scenarios/kubernetes_e2e.py
+      args:
+      - --cluster=gce-scale-cluster
+      - --env=CONCURRENT_SERVICE_SYNCS=5
+      - --env=HEAPSTER_MACHINE_TYPE=e2-standard-32
+      - --extract=ci/latest-fast
+      - --extract-ci-bucket=k8s-release-dev
+      # Overrides CONTROLLER_MANAGER_TEST_ARGS from preset-e2e-scalability-periodics.
+      - --env=CONTROLLER_MANAGER_TEST_ARGS=--profiling --kube-api-qps=100 --kube-api-burst=100 --endpointslice-updates-batch-period=500ms
```
Comment (mborsz) [inline config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml:31]: Value of this flag has changed in https://github.com/kubernetes/test-infra/pull/22446

**Comment by jkaniuk** [comment]
There are various GCP quotas apart from CPU one that need to be lifted to run 5k node jobs (to be copied from legacy project).

Are you tracking this anywhere?

**Comment by ameukam** [comment]
> There are various GCP quotas apart from CPU one that need to be lifted to run 5k node jobs (to be copied from legacy project).
> 
> Are you tracking this anywhere?

@jkaniuk For  the moment nothing is tracked. Do you have anexisting document with the list of quotas that need to be raised ? I can create a new document if not.

**Comment by jkaniuk** [comment]
> @jkaniuk For the moment nothing is tracked. Do you have an existing document with the list of quotas that need to be raised ? I can create a new document if not.

Please do, I do not believe we have anything yet.

@wojtek-t, @jprzychodzen

**Comment by ameukam** [comment]
> > @jkaniuk For the moment nothing is tracked. Do you have an existing document with the list of quotas that need to be raised ? I can create a new document if not.
> 
> Please do, I do not believe we have anything yet.
> 
> @wojtek-t, @jprzychodzen

@jkaniuk I created https://docs.google.com/spreadsheets/d/1v9ynsUx3pcMJKVaHViuVqxVtRP1OmGIadvw2uro7D7Y/edit#gid=0. Feel free to adjust at your own convenience.

File: config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml (hunk 4 above)
Comment (ameukam) [reply config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml]: Fixed.

File: config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml (hunk 1 above)
Comment (ameukam) [reply config/jobs/kubernetes/wg-k8s-infra/wg-k8s-infra-canaries.yaml:35]: @spiffxp I'll make the quota requests in `k8s-infra-e2e-scale-5k-project`. See https://github.com/kubernetes/k8s.io/issues/2225

**Comment by dims** [comment]
/lgtm
/hold

please feel free to remove hold if you are ready.

**Pull request merged**